src/pages/User.js

```diff
@@ -0,0 +1,219 @@
+  });
+  const [updateInfoBtn, setUpdateInfoBtn] = useState({
+    isDisabled: true,
+    isLoading: false,
+  });
+  const [changePasswordBtn, setChangePasswordBtn] = useState({
+    isDisabled: true,
+    isLoading: false,
+  });
+
+  const handleUpdateUser = async (e) => {
+    e.preventDefault();
+    setUpdateInfoBtn((state) => ({
+      ...state,
+      isDisabled: true,
+      isLoading: true,
+    }));
+    if (userCredentials.email.length > 0) {
+      await updateUser(userCredentials.email);
+      setUserCredentials((credentials) => ({ ...credentials, email: "" }));
+    }
+    setUpdateInfoBtn((state) => ({
+      ...state,
+      isDisabled: true,
+      isLoading: false,
+    }));
+  };
+
+  const handleChangePassword = async (e) => {
+    e.preventDefault();
+    setChangePasswordBtn((state) => ({
+      ...state,
+      isDisabled: true,
+      isLoading: true,
+    }));
+    await changePassword(
+      userCredentials.oldPassword,
+      userCredentials.confirmPassword
+    );
+    setChangePasswordBtn((state) => ({
+      ...state,
+      isDisabled: true,
+      isLoading: false,
+    }));
+    setUserCredentials((state) => ({
+      ...state,
+      oldPassword: "",
+      newPassword: "",
+      confirmPassword: "",
+    }));
+  };
+
+  const handleEmailInput = (e) => {
+    setUserCredentials((credentials) => ({
+      ...credentials,
+      email: e.target.value,
+    }));
+    if (e.target.value.length > 0) {
+      setUpdateInfoBtn((state) => ({ ...state, isDisabled: false }));
+    } else {
+      setUpdateInfoBtn((state) => ({ ...state, isDisabled: true }));
+    }
+  };
+
+  const handleOldPasswordInput = (e) => {
+    setUserCredentials((credentials) => ({
+      ...credentials,
+      oldPassword: e.target.value,
+    }));
+  };
+
+  const handleNewPasswordInput = (e) => {
+    setUserCredentials((credentials) => ({
+      ...credentials,
+      newPassword: e.target.value,
+    }));
+  };
+
+  const handleConfirmPasswordInput = (e) => {
+    setUserCredentials((credentials) => ({
+      ...credentials,
+      confirmPassword: e.target.value,
+    }));
+    if (userCredentials.newPassword === e.target.value) {
+      setChangePasswordBtn((state) => ({
+        ...state,
+        isDisabled: false,
+      }));
+    } else {
+      setChangePasswordBtn((state) => ({
+        ...state,
+        isDisabled: true,
+      }));
+    }
+  };
+
+  if (isUserLoading) {
+    return (
+      <DefaultWithoutSearch>
+        <h1 className="overlay">Loading ...</h1>
+      </DefaultWithoutSearch>
+    );
+  }
+
+  return (
+    <DefaultWithoutSearch>
+      <div className="form--container justify-space">
+        <form className="form--control" onSubmit={handleUpdateUser}>
+          <h1>Update Info</h1>
+          <label htmlFor="email">
+            <div>Username</div>
+            <input value={user.username} type="text" name="username" disabled />
+          </label>
+          <label htmlFor="email">
```

Comment:
`htmlFor="email"` htmlFor expects id, it will not work here (clicking on the label should focus input).

May try to wrap input inside a label that this `htmlFor` is not required. Use CSS to place it properly 
```
<label>
  Text:
  <input />
</label>
```